quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -183,11 +183,15 @@ impl Storage for LocalFileStorage {
         if let Some(parent_dir) = full_path.parent() {
             fs::create_dir_all(parent_dir).await?;
         }
-
         let mut reader = payload.byte_stream().await?.into_async_read();
-        let mut f = tokio::fs::File::create(full_path).await?;
-        tokio::io::copy(&mut reader, &mut f).await?;
-
+        let named_temp_file = tempfile::NamedTempFile::new_in(&self.root)?;
+        let (temp_std_file, temp_filepath) = named_temp_file.into_parts();
+        let mut temp_tokio_file = tokio::fs::File::from_std(temp_std_file);
+        tokio::io::copy(&mut reader, &mut temp_tokio_file).await?;
+        temp_tokio_file.flush().await?;
```

Comment:
Is the intention here to make sure the file gets dropped immediately or is this to ensure data is correctly persisted to disk?

 If it's the latter we should use `sync_data` or `sync_all`.

Reply:
the main intention is to make sure readers don't read a partially written file. Adding an fsync is not a bad idea.

Reply:
added a `sync_data` on the file and the parent directory.

@chillfish8 can you resume the review?